test/battle/ai.c

```diff
@@ -0,0 +1,413 @@
+#include "global.h"
+#include "test/battle.h"
+#include "battle_ai_util.h"
+
+AI_BATTLE_TEST("AI gets baited by Protect Switch tactics") // This behavior is to be fixed.
+{
+    GIVEN {
+        AI_FLAGS(AI_FLAG_CHECK_BAD_MOVE | AI_FLAG_CHECK_VIABILITY | AI_FLAG_TRY_TO_FAINT | AI_FLAG_SMART_SWITCHING);
+        PLAYER(SPECIES_STUNFISK);
+        PLAYER(SPECIES_PELIPPER);
+        OPPONENT(SPECIES_DARKRAI) { Moves(MOVE_TACKLE, MOVE_PECK, MOVE_EARTHQUAKE, MOVE_THUNDERBOLT); }
+        OPPONENT(SPECIES_SCIZOR) { Moves(MOVE_HYPER_BEAM, MOVE_FACADE, MOVE_GIGA_IMPACT, MOVE_EXTREME_SPEED); }
+    } WHEN {
+
+        TURN { MOVE(player, MOVE_PROTECT);  EXPECTED_MOVE(opponent, MOVE_EARTHQUAKE); } // E-quake
+        TURN { SWITCH(player, 1);           EXPECTED_MOVE(opponent, MOVE_EARTHQUAKE); } // E-quake
+        TURN { MOVE(player, MOVE_PROTECT);  EXPECTED_MOVE(opponent, MOVE_THUNDERBOLT); } // T-Bolt
+        TURN { SWITCH(player, 0);           EXPECTED_MOVE(opponent, MOVE_THUNDERBOLT); } // T-Bolt
+        TURN { MOVE(player, MOVE_PROTECT);  EXPECTED_MOVE(opponent, MOVE_EARTHQUAKE); } // E-quake
+        TURN { SWITCH(player, 1);           EXPECTED_MOVE(opponent, MOVE_EARTHQUAKE);} // E-quake
+        TURN { MOVE(player, MOVE_PROTECT);  EXPECTED_MOVE(opponent, MOVE_THUNDERBOLT); } // T-Bolt
+    } SCENE {
+    }
```

Comment:
We don't have to write empty `SCENE`s, and I would expect most AI tests to not need them :)
```suggestion
    }
```